### PR TITLE
Adding To Core to the Easy Pipeline

### PR DESCRIPTION
**Pull request by mariari**

### CODE REVIEW

File: library/EasyPipeline/src/Easy.hs [hunk 1]
```diff
@@ -36,59 +39,77 @@ import qualified Juvix.Frontend.Types.Base as Frontend
...
+    param :: Core.Parameterisation primTy primVal
   }
   deriving (Show)
 
 -- we can override defaults by saying def { newOptions }
-def :: Options
+def :: Options primTy primVal
 def =
   Opt
     { -- to avoid being overhwlemed in the repl by giant text, we have
```
Comment (systemf-frank) [inline library/EasyPipeline/src/Easy.hs:70]: overhwlemed -> overwhelmed

File: library/EasyPipeline/src/Easy.hs [hunk 2]
```diff
@@ -36,59 +39,77 @@ import qualified Juvix.Frontend.Types.Base as Frontend
...
           "../../stdlib/Michelson.ju",
           "../../stdlib/MichelsonAlias.ju"
-        ]
+        ],
+      param = Michelson.Param.michelson
     }
 
 -- @defCircuit@ gives us the circuit prelude
-defCircuit :: Options
+-- defCircuit :: Options
```
Comment (systemf-frank) [inline library/EasyPipeline/src/Easy.hs:93]: Why no type specification?

File: library/EasyPipeline/src/Easy.hs [hunk 3]
```diff
@@ -188,15 +209,15 @@ desugarMinimalPrelude = desugarLibrary def
 
 -- | @contextifyGen@ is the generator function for the various contexitfy passes
 contextifyGen ::
-  (NonEmpty (NameSymb.T, [Sexp.T]) -> IO b) -> ByteString -> Options -> IO b
+  (NonEmpty (NameSymb.T, [Sexp.T]) -> IO b) -> ByteString -> Options pt pv -> IO b
```
Comment (systemf-frank) [inline library/EasyPipeline/src/Easy.hs:212]: Personally, I would write `primTy` and `primVal` here, to be in line with the rest of the code.

File: library/EasyPipeline/src/Easy.hs [hunk 4]
```diff
@@ -188,15 +209,15 @@ desugarMinimalPrelude = desugarLibrary def
...
 contextifyGen f text def = do
   lib <- desugarLibrary def
   let dusugared = desugar text
   f ((currentContextName def, dusugared) :| lib)
 
 -- | @contextifyFileGen@ is like @contextifyGen@ but for the file variants
 contextifyFileGen ::
-  (NonEmpty (NameSymb.T, [Sexp.T]) -> IO b) -> FilePath -> Options -> IO b
+  (NonEmpty (NameSymb.T, [Sexp.T]) -> IO b) -> FilePath -> Options pt pv -> IO b
```
Comment (systemf-frank) [inline library/EasyPipeline/src/Easy.hs:220]: Same.

File: library/EasyPipeline/src/Easy.hs [hunk 5]
```diff
@@ -314,10 +388,68 @@ printModule name ctx =
...
+  printDefModule option context
+  --
+  let currentDefinedItems = definedFunctionsInModule option context
+  --
+  Right cored <- coreifyFile file option
+  traverse_ (printCoreFunction cored option) currentDefinedItems
+
 printDefModule ::
-  (MonadIO m, Show ty, Show term, Show sum) => Options -> Context.T term ty sum -> m ()
+  (MonadIO m, Show ty, Show term, Show sum) => Options a b -> Context.T term ty sum -> m ()
```
Comment (systemf-frank) [inline library/EasyPipeline/src/Easy.hs:438]: Same again.

File: library/EasyPipeline/src/Easy.hs [hunk 6]
```diff
@@ -36,59 +39,77 @@ import qualified Juvix.Frontend.Types.Base as Frontend
...
 import Prelude (error)
+import qualified Prelude (Show (..))
 
 --------------------------------------------------------------------------------
 -- OPTIONS
 --------------------------------------------------------------------------------
 
-data Options = Opt
+instance Prelude.Show (Core.Parameterisation primTy primVal) where
+  show _ = "param"
```
Comment (systemf-frank) [inline library/EasyPipeline/src/Easy.hs:57]: Should we define a proper instance in Core? Apparently we never had a use for it, so maybe not.

File: library/EasyPipeline/src/Easy.hs [hunk 7]
```diff
@@ -36,59 +39,77 @@ import qualified Juvix.Frontend.Types.Base as Frontend
...
 import qualified Juvix.Library.Feedback as Feedback
+import qualified Juvix.Library.HashMap as Map
 import qualified Juvix.Library.NameSymbol as NameSymb
 import qualified Juvix.Library.Sexp as Sexp
 import qualified Juvix.Pipeline as Pipeline
 import qualified Juvix.Pipeline.Compile as Compile
+import qualified Juvix.ToCore.Types as ToCore.Types
 import qualified Text.Pretty.Simple as Pretty
 import Prelude (error)
+import qualified Prelude (Show (..))
```
Comment (systemf-frank) [inline library/EasyPipeline/src/Easy.hs:50]: Separate from this PR: shouldn't standardlibrary include this?

**Review by systemf-frank — CHANGES_REQUESTED** [review]

File: library/EasyPipeline/src/Easy.hs (hunk 2 above)
Comment (mariari) [reply library/EasyPipeline/src/Easy.hs:93]: Requires another library to be imported IIRC, so I just left this one out

File: library/EasyPipeline/src/Easy.hs (hunk 7 above)
Comment (mariari) [reply library/EasyPipeline/src/Easy.hs:50]: Protolude, has it's own version of `Show` and so if you make this standard, it'd be ambiguities all over the place

Comment (systemf-frank) [reply library/EasyPipeline/src/Easy.hs:50]: Ah OK.

**Review by systemf-frank — APPROVED** [review]

**Pull request merged**